spec/ordinary-and-exotic-object-behaviours.html

```diff
@@ -0,0 +1,322 @@
+        <emu-alg>
+          1. If Type(_P_) is String, then
+            1. Let _numericIndex_ be ! CanonicalNumericIndexString(_P_).
+            1. If _numericIndex_ is not *undefined*, then
+              1. Let _value_ be ? TupleGet(_T_, _numericIndex_).
+              1. If _value_ is ~empty~, return *undefined*.
+              2. Return _value_.
+          1. Let _parent_ be ? _T_.[[GetPrototypeOf]]().
+          1. Return ? _parent_.[[Get]](_P_, _Receiver_).
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-tuple-exotic-objects-set-p-receiver">
+        <h1>[[Set]] ( _P_, _Receiver_ )</h1>
+        <p>When the [[Set]] internal method of a Tuple exotic object _T_ is called with property key _P_ and ECMAScript language value _Receiver_, the following steps are taken:</p>
+        <emu-alg>
+          1. Return *false*.
+        </emu-alg>
+      </emu-clause>
+      
+      <emu-clause id="sec-tuple-exotic-objects-ownpropertykeys">
+        <h1>[[OwnPropertyKeys]] ( )</h1>
+        <p>When the [[OwnPropertyKeys]] internal method of a Record exotic object _R_ is called, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _keys_ be a new empty List.
+          1. Assert _T_ is Tuple exotic object.
+          1. Let _len_ be the length of _T_.[[TupleData]].[[Sequence]].
+          1. For each integer _i_ starting with 0 such that _i_ &lt; _len_ in ascending order, do
+            1. Add ! ToString(_i_) to _keys_.
+          1. Return _keys_
+        </emu-alg>
+      </emu-clause>
+
+      
+      <emu-clause id="sec-tuplecreate" aoid="TupleCreate">
+        <h1>TupleCreate ( _value_ )</h1>
+        <p>This abstract operation TupleCreate takes a _value_ argument. It is used to specify the creation of new Tuple exotic objects. It performs the following steps when called:</p>
+        <emu-alg>
+            1. Assert: Type(_value_) is Tuple.
+            1. Assert: Type(_value_.[[Sequence]]) is List.
+            1. Let _T_ be ! MakeBasicObject(&laquo; [[TupleData]], [[Prototype]] &raquo;).
+            1. Set _T_'s essential internal methods to the definitions specified in <emu-xref href="#sec-tuple-exotic-objects"></emu-xref>.
+            1. Set _T_.[[Prototype]] to %Tuple.prototype%.
+            1. Set _T_.[[TupleData]] to _value_.
+            1. Return _T_
+            
+        </emu-alg>
+      </emu-clause>
+      
+      <emu-clause id="sec-isvalidtupleindex-r-p" aoid="IsValidTupleIndex">
+        <h1>IsValidTupleIndex ( _T_, _numericIndex_ )</h1>
+        <p>This abstract operation IsValidTupleIndex takes arguments _T_ and _numericIndex_. It performs the following steps when called:</p>
+        <emu-alg>
+          1. Assert: _T_ is an Object that has [[TupleData]] internal slot.
+          1. Assert: Type(_numericIndex_) is an integer.
```

Comment:
"integer" isn't a type. you may want this:
```suggestion
          1. Assert: ! IsInteger(_numericIndex_) is *true*.
```